### ISSUE TITLE
[macOS] Crashes when focus loss from dialog on macOS 10.14 (again)

### ISSUE THREAD

**Issue by ssaring** [javafxports/openjdk-jfx#480]
There was already a bug on macOS 10.14 that causes the JavaFX application to crash when there is an opened JavaFX dialog and the user switches between the JavaFX application and other applications.

This was the previous bug https://bugs.openjdk.java.net/browse/JDK-8211304 and it has been fixed in OpenJFX 12 and backported to 8 and 11.0.2.

Unfortunately this bug occurs again after updating my JavaFX application to JDK 12 and OpenJFX 12, maybe this happens since macOS has been updated to version 10.14.5 (can't test it with previous versions anymore).

The bug can be reproduced with the same HelloWorld application included in the previous bug JDK-8211304.

This is the console output after the application crashes:

> 2019-05-23 21:13:30.716 java[17311:204875] unrecognized type is 4294967295
> 2019-05-23 21:13:30.716 java[17311:204875] *** Assertion failure in -[NSEvent _initWithCGEvent:eventRef:], /BuildRoot/Library/Caches/com.apple.xbs/Sources/AppKit/AppKit-1671.50.111/AppKit.subproj/NSEvent.m:1977
> 2019-05-23 21:13:30.717 java[17311:204875] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Invalid parameter not satisfying: _type > 0 && _type <= kCGSLastEventType'
> *** First throw call stack:
> 
> (
> 0 CoreFoundation 0x00007fff2c136cfd __exceptionPreprocess + 256
> 1 libobjc.A.dylib 0x00007fff567dca17 objc_exception_throw + 48
> 2 CoreFoundation 0x00007fff2c151a1a +[NSException raise:format:arguments:] + 98
> 3 Foundation 0x00007fff2e3f0c85 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 194
> 4 AppKit 0x00007fff2970ebe3 -[NSEvent _initWithCGEvent:eventRef:] + 3296
> 5 AppKit 0x00007fff29a5af53 +[NSEvent eventWithCGEvent:] + 117
> 6 libglass.dylib 0x0000000111252f3b listenTouchEvents + 59
> 7 SkyLight 0x00007fff512e0d5c _ZL19processEventTapDataPvjjjPhj + 157
> 8 SkyLight 0x00007fff511e1430 _XPostEventTapData + 277
> 9 SkyLight 0x00007fff512e0c61 ZL22eventTapMessageHandlerP12__CFMachPortPvlS1 + 132
> 10 CoreFoundation 0x00007fff2c098699 __CFMachPortPerform + 246
> 11 CoreFoundation 0x00007fff2c098597 CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION + 41
> 12 CoreFoundation 0x00007fff2c0984f5 __CFRunLoopDoSource1 + 527
> 13 CoreFoundation 0x00007fff2c0804ec __CFRunLoopRun + 2524
> 14 CoreFoundation 0x00007fff2c07f8be CFRunLoopRunSpecific + 455
> 15 Foundation 0x00007fff2e2e38ef -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 280
> 16 libglass.dylib 0x00000001112479f5 +[GlassApplication enterNestedEventLoopWithEnv:] + 165
> 17 libglass.dylib 0x000000011124848a Java_com_sun_glass_ui_mac_MacApplication__1enterNestedEventLoopImpl + 74
> 18 ??? 0x0000000111a5e990 0x0 + 4591053200
> )
> libc++abi.dylib: terminating with uncaught exception of type NSException
> 
> Process finished with exit code 134 (interrupted by signal 6: SIGABRT)

Here are my system and library informations:

* macOS 10.14.5
* AdoptOpenJDK (build 12.0.1+12)
* OpenJFX 12.0.1 (also tested with 11.0.2 and 13-ea+8, it crashes with all of them)

Are there any other macOS users which have the same problems and maybe a workaround / fix?

**Comment by ssaring** [comment]
Does it make sense to reopen the previous bug JDK-8211304 ? I have no permission for that.

**Comment by kevinrushforth** [comment]
No, we never reopen a bug that is resolved/fixed with a committed changeset. Please file a new bug. Please include the exact command line that you use to run it and any other information, such as the hs_err*.log file and the macOs crash log.

**Comment by kevinrushforth** [comment]
Can you also double-check where the library is loaded from? The output of `java -Djavafx.verbose=true` would be helpful. Thanks.

**Comment by kevinrushforth** [comment]
I took a quick look at the code, and I don't see how this exception is possible if you are running a version of FX that includes the fix for [JDK-8211304](https://bugs.openjdk.java.net/browse/JDK-8211304). So one possibility is that it is picking up an old version of libglass.dylib from somewhere, thus the request for the output of `-Djavafx.verbose=true` and the contents of the `hs_err*.log` file.

**Comment by kevinrushforth** [comment]
I tested this on macOS 10.14.5 and it works fine for me. Further, the following in the stack trace is the exact same offset in the `listenTouchEvents` function as the original crash:

```
 6 libglass.dylib 0x0000000111252f3b listenTouchEvents + 59
```

Given this, I am fairly certain that you are somehow running an older version of libglass.dylib without the fix. Closing this as incomplete pending more information.

**Comment by ssaring** [comment]
@kevinrushforth Thanks a lot for your hints and ideas, you are right - an older version of JavaFX is used in parallel with the new version.

And I've found the reason for this weird behavior - it's the ControlsFX library which I've updated recently from 9.0.0 to the new released version 11.0.0. This ControlsFX version contains dependencies to OpenJFX version 11, which contains the bug, see https://search.maven.org/artifact/org.controlsfx/controlsfx/11.0.0/jar
In the previous version there were no dependencies, that's why it worked without problems.

So my application contains the OpenJFX 11 (from controlsfx)  and 12.0.1 (in my applications POM file) in the classpath and that's why it crashes.
This means that this bug can occur for all macOS users which are using an JavaFX application with ControlsFX 11.0.0 as an dependency. I will file a bug report to the ControlsFX project for that.

Once again, thanks a lot for your help!
